Feature database

jitendra2329 opened the pull request:
PostgreSQL database is added int the project.
CRUD operations are defined on the database.

jitendra2329 committed:
README.md is added

jitendra2329 committed:
README.md file is added

jitendra2329 committed:
database is being added in the project and CRUD operations are defined on the database

Bhavya-verma-knoldus commented on src/main/scala/App/Main.scala:
```diff
@@ -2,69 +2,132 @@ package App
+    val typeOfUser: Either[Unit, UserType] = userType match {
+      case "Admin" => Right(Admin)
+      case "Customer" => Right(Customer)
+      case _ => Left("Invalid user")
```
You are returning String here. So the return type should be Either[String, UserType]

Bhavya-verma-knoldus commented on src/main/scala/Dao/DAO.scala:
```diff
@@ -1,21 +1,21 @@
   def addUser(user: Users): Future[String]
 
-  def getById(userId: UUID): Future[Option[Users]]
+  def getById(userId: Int): Future[ResultSet]
```
Why are you using Int type here? Why not UUID?

jitendra2329 replied:
I have used Int because DB is returning Integer Value.

Bhavya-verma-knoldus commented on src/main/scala/db/UserDB.scala:
```diff
@@ -2,47 +2,62 @@ package db
   }
 
-  def getById(userId: UUID): Future[Option[Users]] = Future {
+  def getById(userId: Int): Future[ResultSet] = Future {
```
DB layer should not return ResultSet. It should return the types like List, Seq.

jitendra2329 replied:
Ok, got it, and UUID may be left by mistake, will remove it

jitendra2329 replied:
okay, copy that.

Bhavya-verma-knoldus commented on src/main/scala/db/UserDB.scala:
```diff
@@ -2,47 +2,62 @@ package db
+    val query = "SELECT * FROM userTable;"
     Thread.sleep(150)
-    userStorage
+    Try(statement.executeQuery(query)) match {
```
Since this code is duplicated from line 27, move this to its own method and call that method here and on line 27.

Bhavya-verma-knoldus commented on src/main/scala/service/UserRepo.scala:
```diff
@@ -2,30 +2,42 @@ package service
-  def getAll: Future[ListBuffer[Users]] = {
-    userDB.getAll
+  def getAll: Future[ResultSet] = {
+    Try(userDB.getAll) match {
```
Code is duplicated here also. So move this also to the method and call that method here and on line 20.

jitendra2329 replied:
It is not in my code(on my local), I have changed it to Future[List[Users]].

jitendra2329 replied:
Oh, okay sorry you mean the duplicacy, I will resolve it.

jitendra2329 committed:
Test case are added and some changes are being make according to the test cases and standard of the code.

Bhavya-verma-knoldus commented on src/main/scala/Dao/DAO.scala:
```diff
@@ -1,21 +1,20 @@
   def addUser(user: Users): Future[String]
 
-  def getById(userId: UUID): Future[Option[Users]]
+  def getById(userId: Int): Future[List[Any]]
```
You should never return `Any` as it can introduce bugs in the code.

Bhavya-verma-knoldus replied:
Why return type is still `Int` and not `UUID`?

Bhavya-verma-knoldus commented on src/main/scala/Dao/DAO.scala:
```diff
@@ -1,21 +1,20 @@
+  def getById(userId: Int): Future[List[Any]]
 
-  def getAll: Future[ListBuffer[Users]]
+  def getAll: Future[List[Any]]
```
Ditto: [ref](https://github.com/jitendra2329/user-management-service/pull/7/files#r1200047964)

jitendra2329 committed:
return type of getById and getAll methods is changed from Future[List[Any]] to Future[List[Users]] and test case are modified accordingly.

jitendra2329 committed:
defined a method getResult for removing duplicate code.

jitendra2329 committed:
Main.scala file modified

Bhavya-verma-knoldus commented on src/main/scala/Dao/DAO.scala:
```diff
@@ -1,21 +1,20 @@
   def addUser(user: Users): Future[String]
 
-  def getById(userId: UUID): Future[Option[Users]]
+  def getById(userId: Int):   Future[List[Users]]
```
This should be an `Future[Option[User]]` as you are returning only a single user.

jitendra2329 committed:
getById and getAll methods modified.

The return type of getById Method is changed from Future[List[Users]] to Future[Option[Users]].

One more method is added in the UserDB for the execution of the queries.
